reason-react-navigation-example/src/TwitterNavigation.re

```diff
@@ -0,0 +1,628 @@
+        <Text style=Style.(style(~color="#444", ()))>
+          {title |> React.string}
+        </Text>
+      </TouchableOpacity>;
+  };
+
+  let contentComponent = (props: ContentComponent.props(unit)) => {
+    let navigation = props->ContentComponent.navigation;
+    <SafeAreaView style={Style.style(~flex=1., ())}>
+      <View
+        style=Style.(
+          style(~borderBottomWidth=1., ~borderBottomColor="#eee", ())
+        )>
+        <Image
+          style=Style.(
+            style(
+              ~backgroundColor="grey",
+              ~width=36.->dp,
+              ~borderRadius=18.,
+              ~margin=11.->dp,
+              ~height=36.->dp,
+              (),
+            )
+          )
+          source=Icons.avatar
+        />
+      </View>
+      <View
+        style=Style.(
+          style(~borderBottomWidth=1., ~borderBottomColor="#eee", ())
+        )>
+        <NavRow route="Profile" title="Profile" navigation />
+        <NavRow route="Lists" title="Lists" navigation />
+        <NavRow route="Bookmarks" title="Bookmarks" navigation />
+        <NavRow route="Moments" title="Moments" navigation />
+      </View>
+      <View
+        style=Style.(
+          style(~borderBottomWidth=1., ~borderBottomColor="#eee", ())
+        )>
+        <NavRow route="Settings" title="Settings and privacy" navigation />
+        <NavRow route="Help" title="Help Centre" navigation />
+      </View>
+    </SafeAreaView>;
+  };
+
+  let navigator =
+    makeWithConfig(
+      {
+        "Root":
+          TabNavigator.(
+            Bottom.makeWithConfig(
+              {
+                "Timeline": Timeline.navigator,
+                "Search": Search.navigator,
+                "Notifications": Notifications.navigator,
+                "Messages": Messages.navigator,
+              },
+              config(~initialRouteName="Timeline", ()),
+            )
+          ),
+      },
+      config(~contentComponent, ()),
+    );
+
+  navigator->setDynamicNavigationOptions(params => {
+    let navigation = params->NavigationParams.navigation;
+    let state = navigation->Navigation.state;
+    let routeName = getActiveRoute(state)->NavigationState.routeNameGet;
+    let swipeEnabled = routeName == "TimelineList";
+
+    NavigationOptions.(t(~tabBarVisible=false, ~swipeEnabled, ()));
+  });
+};
+
+module ContentView = {
+  let navigator =
+    TabNavigator.(
+      Material.make({
+        "Content": Content.navigator,
+        "Camera": Camera.navigator,
+      })
+    );
+  let header = header(`Null(Js.null));
+  navigator->setNavigationOptions(NavigationOptions.t(~header, ()));
+};
+
+let app =
+  AppContainer.make(
+    SwitchNavigator.make({
+      "App":
+        StackNavigator.(
+          makeWithConfig(
+            {
+              "View": ContentView.navigator,
+              "Compose": Compose.make,
+              "Tweet": Tweet.make,
+              "Profile": Profile.make,
+            },
```

Comment:
Could we have in mind a safer way than strings to reference screens? I always found that totally unsafe when using rrnav in js...

Reply:
Absolutely. Currently still in the "making it work" phase :) Will look at this next.